src/api/list.c

```diff
@@ -308,59 +351,115 @@ static int api_list_write(struct ftl_conn *api,
 	{
 		// Test validity of this regex
 		regexData regex = { 0 };
-		okay = compile_regex(row.item, &regex, &regex_msg);
-	}
-
-	// Try to add item to table
-	const char *sql_msg = NULL;
-	if(okay && (okay = gravityDB_addToTable(listtype, &row, &sql_msg, api->method)))
-	{
-		if(listtype != GRAVITY_GROUPS)
+		cJSON *it = NULL;
+		cJSON_ArrayForEach(it, row.items)
 		{
-			cJSON *groups = cJSON_GetObjectItemCaseSensitive(api->payload.json, "groups");
-			if(groups != NULL)
-				okay = gravityDB_edit_groups(listtype, groups, &row, &sql_msg);
-			else
-				// The groups array is optional, we still succeed if it
-				// is omitted (groups stay as they are)
-				okay = true;
+			// If any element isn't a string, break early
+			if(!cJSON_IsString(it))
+			{
+				okay = false;
+				break;
+			}
+
+			// Check every array element for its validity
+			okay = compile_regex(it->valuestring, &regex, &regex_msg);
+
+			// TODO: The regex needs to be freed here
```

Comment:
Do you want to do that as part of this PR?

Reply:
Done